SofaKernel/modules/SofaCore/src/sofa/core/topology/Topology.h

```diff
@@ -19,143 +19,81 @@
-    {
-    public:
-        Edge(): sofa::helper::fixed_array<PointID,2>(Topology::InvalidID, Topology::InvalidID){}
-        Edge(PointID a, PointID b) : sofa::helper::fixed_array<PointID,2>(a,b) {}
-    };
-
-    class Triangle : public sofa::helper::fixed_array<PointID,3>
-    {
-    public:
-        Triangle(): sofa::helper::fixed_array<PointID,3>(Topology::InvalidID, Topology::InvalidID, Topology::InvalidID) {}
-        Triangle(PointID a, PointID b, PointID c) : sofa::helper::fixed_array<PointID,3>(a,b,c) {}
-    };
-
-    class Quad : public sofa::helper::fixed_array<PointID,4>
-    {
-    public:
-        Quad(): sofa::helper::fixed_array<PointID,4>(Topology::InvalidID, Topology::InvalidID, Topology::InvalidID, Topology::InvalidID) {}
-        Quad(PointID a, PointID b, PointID c, PointID d) : sofa::helper::fixed_array<PointID,4>(a,b,c,d) {}
-    };
-
-    class Tetrahedron : public sofa::helper::fixed_array<PointID,4>
-    {
-    public:
-        Tetrahedron(): sofa::helper::fixed_array<PointID,4>(Topology::InvalidID, Topology::InvalidID, Topology::InvalidID, Topology::InvalidID) {}
-        Tetrahedron(PointID a, PointID b, PointID c, PointID d) : sofa::helper::fixed_array<PointID,4>(a,b,c,d) {}
-    };
-    typedef Tetrahedron                         Tetra;
-
-    class Pyramid : public sofa::helper::fixed_array<PointID,5>
-    {
-    public:
-        Pyramid(): sofa::helper::fixed_array<PointID,5>(Topology::InvalidID, Topology::InvalidID, Topology::InvalidID, Topology::InvalidID, Topology::InvalidID) {}
-        Pyramid(PointID a, PointID b, PointID c, PointID d, PointID e) : sofa::helper::fixed_array<PointID,5>(a,b,c,d,e) {}
-    };
-
-    class Pentahedron : public sofa::helper::fixed_array<PointID,6>
-    {
-    public:
-        Pentahedron(): sofa::helper::fixed_array<PointID,6>(Topology::InvalidID, Topology::InvalidID, Topology::InvalidID, Topology::InvalidID, Topology::InvalidID, Topology::InvalidID) {}
-        Pentahedron(PointID a, PointID b, PointID c, PointID d, PointID e, PointID f) : sofa::helper::fixed_array<PointID,6>(a,b,c,d,e,f) {}
-    };
-    typedef Pentahedron                          Penta;
-
-    class Hexahedron : public sofa::helper::fixed_array<PointID,8>
-    {
-    public:
-        Hexahedron(): sofa::helper::fixed_array<PointID,8>(Topology::InvalidID, Topology::InvalidID, Topology::InvalidID, Topology::InvalidID,
-                                                           Topology::InvalidID, Topology::InvalidID, Topology::InvalidID, Topology::InvalidID) {}
-        Hexahedron(PointID a, PointID b, PointID c, PointID d,
-                   PointID e, PointID f, PointID g, PointID h) : sofa::helper::fixed_array<PointID,8>(a,b,c,d,e,f,g,h) {}
-    };
-    typedef Hexahedron                          Hexa;
-
-
-
     SOFA_CLASS(Topology, core::objectmodel::BaseObject);
     SOFA_BASE_CAST_IMPLEMENTATION(Topology)
+
+    using Index = sofa::topology::geometry::Index;
```

Comment:
this is for backward compatibility right?

Reply:
yessir
I tried to remove Topology and/or remove its BaseObject inheritance and thats too much a mess for now. We can see that later on if you want